specs/WinRT/MakeMSIX-spec.md

```diff
@@ -0,0 +1,1002 @@
+# MakeMSIX API in Windows App SDK
+- [1. Background](#1-background)
+- [2. Examples](#2-examples)
+    - [2.1. Packing a package](#21-packing-a-package)
+    - [2.2. Unbundling and unpacking a package to change information](#22-unbundling-and-unpacking-a-package-to-change-information)
+    - [2.3. Helper methods for examples](#23-helper-methods-for-examples) 
+- [3. API Details](#3-API-details)
+- [4. Appendix](#4-Appendix)
+- [5. Existing interfaces and options](#5-existing-interfaces-and-options)
+  - [5.1. Makeappx](#51-makeappx)
+    - [5.1.1. pack](#511-pack)
+    - [5.1.2. bundle](#512-bundle)
+    - [5.1.3. unpack](#513-unpack)
+    - [5.1.4. unbundle](#514-unbundle)
+  - [5.2. Makemsix](#52-makemsix)
+    - [5.2.1. pack](#521-pack)
+    - [5.2.2. bundle](#522-bundle)
+    - [5.2.3. unpack](#523-unpack)
+    - [5.2.4. unbundle](#524-unbundle)
+  - [5.3. MSIXMgr](#53-msixmgr)
+    - [5.3.1. unpack](#531-unpack)
+- [6. Summary of existing options in relation to proposed API](#6-summary-of-existing-options-in-relation-to-proposed-API)
+
+# 1. Background
+
+The MakeMSIX API allows developers to create app packages for distribution. 
+The API provides a WinRT interface that supports:
+1. Creation of msix packages and msixbundle bundle packages,
+2. Unpacking and unbundling of packages
+3. Creation of vhd\vdx\cim images for Azure App Attach. 
+4. Creation of Kozani packages.
+
+Creation of Kozani packages is a new feature that has not existed in other msix packaging tools. The
+other features have previously been available in command line tools discussed in the appendix.
+
+Kozani packages are msix packages that have been optimized for size by removing unnecessary
+resources. All existing tools that work with msix packages are compatible with Kozani packages.
+Kozani packages can be created manually without this API by modifying the files, manifest, and
+resource index of an existing package using existing win32 apis.
+
+# 2. Examples
+
+## 2.1. Packing a package
+
+```cpp
```

Comment:
Samples should be written in C# unless there's reasons another language is needed (C++ specific details, not callable by C#, etc)

RECOMMEND: Change the samples to C# e.g.

```
void CreatePackagesFromFolderExample()
{
     // Scenario: ...
     ....
     string packageLayoutRootPath = "D:\\test\\ContosoApp1\\BuildOutput\\PackageLayouts";
    ...
}
```